cmd/agent/app/server.go

```diff
@@ -133,6 +135,31 @@ func (a *Agent) runProxyConnection(o *options.GrpcProxyAgentOptions, drainCh, st
 		}),
 	}
 	cc := o.ClientSetConfig(dialOptions...)
+
+	if o.ServerLeaseSelector != "" {
+		var k8sClient *kubernetes.Clientset
+		config, err := clientcmd.BuildConfigFromFlags("", o.KubeconfigPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load kubernetes client config: %v", err)
+		}
+		k8sClient, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create kubernetes clientset: %v", err)
+		}
+		serverLeaseSelector, err := labels.Parse(o.ServerLeaseSelector)
```

Comment:
This should never happen. Maybe we should be caching the result of the parse we called during validation?

Reply:
Could we just swallow the error instead? Would require less passing state around.